xml_utils.c

```diff
@@ -88,7 +88,7 @@ char **get_xml_list(char * buf_dir_name, int *list_size)
 		return NULL;
 	}
 
-	while (ent = readdir(d), ent != NULL) {
+	while ((ent = readdir(d))) {
```

Comment:
The extra parenthesis are probably not needed.

Reply:
seems it's needed based on default GCC options;
build fails with:
```
/home/travis/build/analogdevicesinc/iio-oscilloscope/xml_utils.c: In function ‘get_xml_list’:
/home/travis/build/analogdevicesinc/iio-oscilloscope/xml_utils.c:91:2: error: suggest parentheses around assignment used as truth value [-Werror=parentheses]
  while (ent = readdir(d)) {
  ^
```

will add it back